### ISSUE TITLE
Cascading route configs

### ISSUE THREAD

**Issue by djadomi** [bcosca/fatfree#973]
[Apologies in advance for long-winded description...]

I've got an app that sets up a default set of "sensible" configs, which can be overridden by site-specific configs. This can also be overridden by host-specific configs, so the dev version of a site has its own config too (if it wants).

```
config/config.ini
config/local/config.ini
config/local/dev.example.com.ini
```

This is partly to have things like

```
[api]
endpoint = https://staging.api.example.com/
```

overridden by

```
[api]
endpoint = https://api.example.com/
```

and also DEBUG, language and so on…

This is all fine for the configs - works great. But for routes it doesn't override the same way. The global `routes.ini` has

```
GET /products = Controller\Products->list
GET /products/@section = Controller\Products->list
GET /products/@section/@product = Controller\Products->list
```

which works fine, but I'm developing a simpler site that doesn't need sections, so this route would be suitable (in `config/local/routes.ini`):

```
GET /products/@product = Controller\Products->list
```

but what's being set is `@PARAMS.section` from the global routes. For now I'm using a horrible hack in the template: `{{@PARAMS.product = @PARAMS.section}}`.

After all that, my question is: is there a way to remove or override an existing route?

**tl;dr:**
Can a route set like `/a/@b` be overridden by another like `/a/@c` to fill the correct element of PARAMS?

**Comment by ikkez** [comment]
please use our [google board](https://groups.google.com/forum/#!forum/f3-framework) for questions like this about architecture and how-to's, because it's a better place for discussions and there are multiple ways to fix your issue, but it's not an issue in F3's core IMO.

**Comment by djadomi** [comment]
OK, thanks for heads-up.
